### ISSUE TITLE
[v1.2 only] Crash on changing graphics page after changing PCjr/Tandy video memory size

### ISSUE THREAD

**Issue by 32bitkid** [robhagemans/pcbasic#102]
## Bug report

**Problem**
When using pcjr/tandy graphics modes and using `clear` to clear video memory *while on a graphical screen*, pcbasic does not reset the video mode and will crash if you attempt to access the newly available screen page using `screen 4,,1,1` (Rather than returning `Illegal function call` like it would before the `clear` command)

On an actual PCjr, when the `clear` command is *not* a noop, it resets the screen to mode `0`. I do not have a Tandy available for testing.

**Program**
```
$ pcbasic --preset=pcjr

10 screen 4
20 clear ,,,32768!
30 screen 4,,1,1
```

**Crash log**
```
EXCEPTION                                                                       
:SCREEN 4,0,1,1                                                                                                                                                 
pcbasic.statements:169, parse_statement                                         
    None                                                                        
pcbasic.statements:2501, exec_screen                                            
    None                                                                        
pcbasic.backend:1040, screen                                                    
    None                                                                        
pcbasic.backend:1187, set_page                                                  
    None                                                                        
IndexError: list index out of range                                             
```

**Notes**

PC-BASIC version:  1.2.14 and 2.0.2beta
Operating system version:  MacOS 10.14.5

**Comment by robhagemans** [comment]
Thank you for reporting this! Looks like there is an issue with keeping track of the video page when the number of pages changes.

Also thanks especially for confirming PCjr behaviour, as I only have access to emulators...

**Comment by robhagemans** [comment]
Reproduced on both 1.2 and 2.0 branches

**Comment by robhagemans** [comment]
Tried this on DOSBox with Tandy emulation and Tandy GW-BASIC and it hangs, so unclear what real Tandy hardware does here. I'm going with dropping to text as with the pcjr

**Comment by robhagemans** [comment]
This is now fixed on `develop` for next release on 2.0 branch, hence marking as 1.2 only

**Comment by robhagemans** [comment]
Fixed on `release-1.2` branch by commit 42983901
